artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+	}
+	return s, nil
+}
+
+// OauthSettings represents the OAuth settings in Artifactory Security Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-Security(Generalsecurity,PasswordPolicy,LDAP,SAML,OAuth,HTTPSSO,Crowd)
+type OauthSettings struct {
+	AllowUserToAccessProfile *bool                  `yaml:"allowUserToAccessProfile,omitempty" xml:"allowUserToAccessProfile,omitempty"`
+	EnableIntegration        *bool                  `yaml:"enableIntegration,omitempty" xml:"enableIntegration,omitempty"`
+	PersistUsers             *bool                  `yaml:"persistUsers,omitempty" xml:"persistUsers,omitempty"`
+	OauthProvidersSettings   *OauthProviderSettings `yaml:"oauthProvidersSettings,omitempty" xml:"oauthProvidersSettings>oauthProvidersSettings,omitempty"`
+	Reset                    *bool                  `yaml:"-" xml:"-"`
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (o OauthSettings) MarshalYAML() (interface{}, error) {
+	if o.Reset != nil && *o.Reset {
+		return nil, nil
+	}
+	return o, nil
+}
+
+// OauthProviderSettings is an alias for a slice of OauthProviderSetting that can be
+// properly marshaled to/from YAML.
+type OauthProviderSettings []*OauthProviderSetting
+
+// MarshalYAML implements the yaml.Marshaller interface for OauthProviderSettings.
+func (o OauthProviderSettings) MarshalYAML() (interface{}, error) {
+	oauthProviderSettingsMap := make(map[string]*OauthProviderSetting)
+	for _, oauthProviderSetting := range o {
+		if *oauthProviderSetting == (OauthProviderSetting{Name: oauthProviderSetting.Name}) {
+			oauthProviderSettingsMap[*oauthProviderSetting.Name] = nil
+		} else {
+			oauthProviderSettingsMap[*oauthProviderSetting.Name] = oauthProviderSetting
+		}
+	}
+
+	return oauthProviderSettingsMap, nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for OauthProviderSettings.
+func (o *OauthProviderSettings) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	oauthProviderSettingsMap := make(map[string]*OauthProviderSetting)
+	if err := unmarshal(oauthProviderSettingsMap); err != nil {
+		return err
+	}
+
+	var oauthProviderSettingsSlice OauthProviderSettings
+	for oauthProviderSettingName, oauthProviderSetting := range oauthProviderSettingsMap {
+		if oauthProviderSetting == nil {
+			continue
+		}
+		oauthProviderSetting.Name = &oauthProviderSettingName
+		oauthProviderSettingsSlice = append(oauthProviderSettingsSlice, oauthProviderSetting)
+	}
+
+	*o = oauthProviderSettingsSlice
+	return nil
+}
+
+// OauthProviderSetting represents the Oauth Provider settings in Artifactory Security Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-Security(Generalsecurity,PasswordPolicy,LDAP,SAML,OAuth,HTTPSSO,Crowd)
+type OauthProviderSetting struct {
+	Name         *string `yaml:"-" xml:"name,omitempty"`
+	Id           *string `yaml:"-" xml:"id,omitempty"`
```

Comment:
I noticed we're not allowing these field to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?

Reply:
Similar to [comment](https://github.com/target/go-arty/pull/57#discussion_r436119180).

Ah, Id shouldn't be `-`... I'll fix that!